File: Programmers/20210106/2.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

bool cmp(const pair<int, string>& a, const pair<int, string>& b) {
    return a.second > b.second;
}

string solution(vector<int> numbers) {
    string answer = "";
    vector <pair<int, string>> v;
    for (int i = 0; i < numbers.size(); i++) {
        v.push_back(make_pair(i, to_string(numbers[i])));
        while (v[i].second.size() <= 4) {
            v[i].second += to_string(numbers[i]);
        }
    }

    sort(v.begin(), v.end(), cmp);

    for (int i = 0; i < v.size(); i++) {
        answer += to_string(numbers[v[i].first]);
    }

    if (answer[0] == '0')
        answer = "0";

    return answer;
}

File: Programmers/Level 2/2.py
def solution(s):
    answer = -1
    stack = []
    for i in s:
        if len(stack) != 0:
            if stack[-1] == i:
                stack.pop()
            else:
                stack.append(i)
        else:
            stack.append(i)

    if len(stack) == 0:
        answer = 1
    else:
        answer = 0
    return answer
